### PR TITLE
fix!: deny majority of window properties

### PR DESCRIPTION
**Pull request by tpoisseau**
and methods.

methodology is `Object.keys(window).sort()` on https://developer.mozilla.org/en-US/docs/Web/API/Window review one by one keys, check doc and choose with my sensibility if we should allow or deny this property from global scope. (remove from list some attributes from external libs like `ga*` from google ads, `__REACT*`)

Please review `window_properties.csv` and confirm if some deny should be allowed and vice-versa.

I Need your opinion about `location`

Closes: https://github.com/cheminfo/eslint-config-cheminfo-react/issues/32

### CODE REVIEW

**Comment by targos** [comment]
To render the csv: https://github.com/cheminfo/eslint-config-cheminfo-react/blob/32-review-all-window-properties-and-improve-restricted-globals/window_properties.csv

**Comment by targos** [comment]
- I think we can omit the event callbacks (`on...`) from the config because it's noisy and unlikely to be a problem for us.
- I would add to the deny list:
  - `alert`, `confirm`, `history`, `location`, `prompt` (too simple names and/or rare usage).
  - `createImageBitmap`, `customElements` (rare usage)
  - `self` (we can use `globalThis` now for Window and Worker compatibility)

Otherwise LGTM.

**Comment by tpoisseau** [comment]
Done,
I let you check

**Comment by tpoisseau** [comment]
```
% npm test

> eslint-config-cheminfo-react@10.1.2 test
> node test/test.js
```

Seems it work, why did you except it should not work ?

**Comment by targos** [comment]
The first array element in a rule should be `'off'`, `'warn'`, or `'error'`.

**Comment by targos** [comment]
The test is probably not good enough. Have a look at https://github.com/cheminfo/eslint-config for a test setup that I know works correctly.

**Comment by tpoisseau** [comment]
https://eslint.org/docs/latest/rules/no-restricted-globals#options
OK, I understood the syntax, which is quite confusing for this rule.

**Comment by tpoisseau** [comment]
So I added small code to check if config is valid.
(I tested it, with invalid rule, it log error and process exit 1. With valid rule, no error)

File: noRestrictedGlobals.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// review: https://github.com/cheminfo/eslint-config-cheminfo-react/pull/33
```
Comment (targos) [inline noRestrictedGlobals.js:1]:
```suggestion
```

Will be linked directly by github

File: eslintrc.test.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+extends:
+  - cheminfo
+  - ./base.js
```
Comment (targos) [inline eslintrc.test.yml:3]:
```suggestion
  - ./base.js
  
```

File: noRestrictedGlobals.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  'setResizable',
+  'speechSynthesis',
+  'status',
+  'statusbar',
+  'stop',
+  'toolbar',
+  'top',
+  'updateCommands',
+  'visualViewport',
+]
```
Comment (targos) [inline noRestrictedGlobals.js:82]:
```suggestion
];

```

File: test/test.js [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
\ No newline at end of file
+// make sure config is valid
+const { ESLint } = require('eslint');
+
+const eslint = new ESLint({ overrideConfigFile: 'eslintrc.test.yml' });
+
+eslint.lintFiles(['test/ok.js']).catch(error => {
+  console.error(error);
+  process.exit(1);
+})
```
Comment (targos) [inline test/test.js:11]:
```suggestion
});

```

**Pull request merged**